use database application_name

chrismeyersfsu opened the pull request:
* add `<hostname>-<pid>-<cmdline>` to application_name in database
* Makes it easier to find the process of a long running query or
long-held lock.
![image](https://user-images.githubusercontent.com/722880/95461283-0c141180-0944-11eb-82d5-5a6552ada0fd.png)

```
SELECT
  pid,
  client_addr,
  application_name,
  now() - pg_stat_activity.query_start AS duration,
  query,
  state
FROM pg_stat_activity;
```

![image](https://user-images.githubusercontent.com/722880/95461351-1fbf7800-0944-11eb-8f6e-faf66e75b019.png)


## TODO:
* truncate `application_name` to 63 characters otherwise when you first establish a connection to the database you will get `NOTICE:  identifier "tower010-vm10-towerperf.xxxxxxx.xxxxxxxx.com-4521-/usr/bin/awx-manage shell_plus" will be truncated to "tower010-vm10-towerperf.usersys.redhat.com-4521-/usr/bin/awx-ma"` To recreate, simply run `awx-manage shell_plus` and run a query, `jt=JobTemplate.objects.count()`
* This code might run too late. The database connection may already be established and we will not get our `application_name` injected. Note, saw this with Gabe and Mellow when debugging

chrismeyersfsu committed:
use database application_name

* add <hostname>-<pid>-<cmdline> to application_name in database
* Makes it easier to find the process of a long running query or
long-held lock.

chrismeyersfsu commented:
recheck

ryanpetrello commented:
@chrismeyersfsu `django.utils.text.Truncator(value).chars(length)`

ryanpetrello reviewed (approved):
If we can get this working 100%, I love the idea of this PR for improving debuggability.

chrismeyersfsu committed:
Revert "use database application_name"

This reverts commit 338f73548ab9491b40fe0a4a5d07996a68849d8a.

chrismeyersfsu committed:
fill in postgres application_name on connection

* Tried to fill in application_name in awx/__init__.py but I think that
is too late
* Fill in database application_name with enough information to easily
trace the connection from postgres back to the node and pid that
initiated the connection.
* Set application_name in django settings so that application_name is
set _before_ the first postgres connection is established.

chrismeyersfsu commented:
> django.utils.text.Truncator(

Wasn't able to use in the `settings.py` implementation because of import before app fully initialized.

ryanpetrello reviewed (approved):
🚢
